### ISSUE TITLE
Use full system version in UserAgent ?

### ISSUE THREAD

**Issue by LudoFran** [dotnet/windows-sdk-for-google-analytics#17]
What do you think of returning the full system version (10.0.14393.693) in the UserAgent instead of only returning 10.0 ?

I think that it can be very interesting to know what build our Users are using.

**Comment by LindaLawton** [comment]
Google Analytics uses the user-agent that the request was sent with to create a number of the metrics.    Device category, OS, version probably a bunch more.    

The [User Agent Override](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ua) can be to change the user agent that was sent with the request.

There is a nice note on that page:

> The User Agent of the browser. Note that Google has libraries to identify real user agents. Hand crafting your own agent could break at any time.

This leads me to believe that the Analytics.js, Android sdk and Ios sdk libraries must correctly create there prospective user-agents.      I think it might be a good idea to ensure that this library is also creating the correct user-agent for the machine the library is running on. 

So basically you can use the override parameter if you like.

**Comment by LudoFran** [comment]
I think that the UserAgent created is correct but it may lack some informations for devs.

Do you know if GA can parse this UserAgent :
`Mozilla/5.0 (Windows NT 10.0.14393.693; Win64; X64; Microsoft Corporation; Surface Pro 4)` 
instead of 
`Mozilla/5.0 (Windows NT 10.0; Win64; X64; Microsoft Corporation; Surface Pro 4)`

Thanks for the tips for User Agent Override, I think that I will use it for now

**Comment by LindaLawton** [comment]
TBH I have never bothered messing with the User-agent the warning put me off a little.   Shouldn't be to hard to test.    I can whip up a quick curl script.

**Comment by LudoFran** [comment]
Arf it can't detect more info...

Well I think that I'll use some CustomVariables then.

Anyway, thank you for your time :)

**Comment by LindaLawton** [comment]
I will check again tomorrow hits can take 24 - 48 hours to process.   So the hit i just sent may be incomplete.     If i don't post back just ping this :)

**Comment by LudoFran** [comment]
I also tried in a dummy project and I never saw the build version. 

Did you have anything @LindaLawton ?

**Comment by LindaLawton** [comment]
@LudoFran  nope all I have been able to get is the full number it shows 10.    I will bring it up with the Google Analytics team next time i chat with them.    It would be interesting to get a full OS version in the analytics especially for apps.     Knowing the full build version of an app that is throwing an error for example would be invaluable.    For now I suggest you just dump it in a custom dimension as you mentioned. 

However as this could be useful I logged it on the issue form for Google Analytics [Issue: OperatingSystemVersion not including full build number](https://issuetracker.google.com/issues/35589653)
